### PR TITLE
Fix overlay plot image

### PR DESCRIPTION
**Pull request by ericpre**
### Progress of the PR
- [x] Tweak figure size to avoid very small size when the signal size is small ,
- [x] update docstring (if appropriate),
- [n/a] update user guide (if appropriate),
- [n/a] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/upcoming_changes/README.rst)),
- [n/a] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [x] add tests,
- [x] ready for review.

### Minimal example of the bug fix or the new feature
```python
import hyperspy.api as hs
import numpy as np
s = hs.signals.Signal2D(np.arange(100).reshape(10, 10))
hs.plot.plot_images([s, s], overlay=True, axes_decor='off')
```
would normally generate a very small figure.

### CODE REVIEW

File: hyperspy/drawing/utils.py [hunk 1]
```diff
@@ -551,13 +550,16 @@ def plot_images(images,
...
     suptitle_fontsize : int, optional
         Font size to use for super title at top of figure.
-    colorbar : 'multi', None, 'single', optional
-        Controls the type of colorbars that are plotted.
+    colorbar : 'default', 'multi', None, 'single', optional
+        Controls the type of colorbars that are plotted, incompatible with
+        ``overlay=True``.
+        If 'default', same as 'multi' when ``overlay=False``, otherwise same
+        as ``None``.
+        If 'multi', individual colorbars are plotted for each
```
Comment (jlaehne) [inline hyperspy/drawing/utils.py]: The following lines are mixed up (end of sentence was not moved).

File: hyperspy/drawing/utils.py [hunk 2]
```diff
@@ -841,22 +854,52 @@ def __check_single_colorbar(cbar):
     else:
         raise ValueError("Did not understand input of labels.")
 
-    # Start of non-overlay?
+    # Check if we need to add a scalebar for some of the images
+    if isinstance(scalebar, (list, tuple)) and all(isinstance(x, int)
+                                                   for x in scalebar):
+        scalelist = True
+    else:
```
Comment (jlaehne) [inline hyperspy/drawing/utils.py:860]: coverage?

**Comment by jlaehne** [comment]
Nice, I noticed that unfavorable behavior a while ago as well. Is there any related issue that would be closed?

**Comment by ericpre** [comment]
> Nice, I noticed that unfavorable behavior a while ago as well. Is there any related issue that would be closed?

I couldn't find one but this is for the overlay, which hasn't been released yet.

File: hyperspy/drawing/utils.py [hunk 3]
```diff
@@ -625,15 +626,17 @@ def plot_images(images,
...
     legend_loc : str, int, optional
         This parameter controls where the legend is placed on the figure
         see the :py:func:`matplotlib.pyplot.legend` docstring for valid values
-    pixel_size_factor : int or float, optional
-        Default value is 1. Sets the size of the figure when plotting an overlay image. The higher
-        the number the larger the figure and therefore a greater number of
-        pixels are used. This value will be ignored if a Figure is provided.
+    pixel_size_factor : None, int or float, optional
+        If ``None`` (default), the size of the figure is taken from the
+        matplotlib ``rcParams``. Otherwise Sets the size of the figure when
```
Comment (jlaehne) [inline hyperspy/drawing/utils.py]:
```suggestion
        matplotlib ``rcParams``. Otherwise sets the size of the figure when
```

**Review by jlaehne — APPROVED** [review]

**Pull request merged**